Account/FirefoxAccount.swift

```diff
@@ -10,6 +10,11 @@ import SwiftyJSON
 
 private let log = Logger.syncLogger
 
+public protocol CollectionChangedNotifier {
+    func notify(deviceIDs: [GUID], collectionsChanged collections: [String]) -> Deferred<Maybe<FxANotifyResponse>>
```

Comment:
I don't think you should return the `FxANotifyResponse`, and I think this interface should be in `Sync`, not `Account`. After all, this is a delegate protocol.